drivers/hmis/app/graphql/mutations/update_form_definition.rb

```diff
@@ -35,15 +35,16 @@ def resolve(id:, input:)
         builder.validate_definition(definition.definition) { |err| errors.add(:definition, message: err) }
       end
 
+      # Return user-facing validation errors for the form content
       return { errors: errors } if errors.present?
 
-      if definition.valid?
-        definition.save!
-        { form_definition: definition }
-      else
-        errors.add_ar_errors(definition.errors)
-        { errors: errors }
-      end
+      # Raise if the definition is not valid (invalid role/status/identifier; not expected)
+      raise "Definition invalid: #{definition.errors.full_messages}" unless definition.valid?
+
+      # Manually save a PaperTrail version, so we know who made the change. Because we `skip` the definition
+      # field, a PaperTrail record will not automatically get created if only the definition changed.
+      definition.paper_trail.save_with_version
```

Comment:
https://www.rubydoc.info/gems/paper_trail/11.0.0/PaperTrail%2FRecordTrail:save_with_version

needed because of https://github.com/greenriver/hmis-warehouse/blob/72dcf983ff7338a4ef85814419e446da8ff71244/drivers/hmis/app/models/hmis/form/definition.rb#L35

This seems to work OK, but we don't use this method anywhere else.

Reply:
using `save_with_version` seems reasonable to me, it's part of paper_trail's documented api